### DISCUSSION TITLE
How to reduce the serialize deserialize time in FastDDS 2.6.7

### DISCUSSION THREAD

**Discussion by Syamkamalnadhc** [eProsima/Fast-DDS#5297] [Q&A]
I am using FastDDS 2.6.7 with ROS2-Humble and I need to optimize the performance, I put performance logs in FastDDS and observed ~300ms delay across the serialize and ~500ms deserialize when sending 1gb data. as per my understanding FastCDR module is handling the serialization process and i need to know are there any tunable configurations to reduce the delay.

**Comment by JesusPoderoso** [comment]
Hi @Syamkamalnadhc, thanks for using Fast DDS.
The serialization (and deserialization) process can be avoided using [loan](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/subscriber/dataReader/readingData.html#loaning-and-returning-data-and-sampleinfo-sequences) mechanism. Also, if your data type and your deployment meet [these requirements](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/zero_copy/zero_copy.html#constraints), the use of [zero-copy](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/zero_copy/zero_copy.html) delivery mechanism may be of interest to improve the performance of the communication and reduce the number of copies performed.

**Reply by Syamkamalnadhc** [reply]
Is the UDP transport supports Loan mechanism?
I got an error "[rclcpp]: Currently used middleware can't loan messages. Local allocator will be used." when using UDPv4 and i did not get this message with SHM transport

**Reply by JesusPoderoso** [reply]
Humble version requires both [data-sharing enabled and plain type](https://github.com/ros2/rmw_fastrtps/blob/humble/rmw_fastrtps_cpp/src/publisher.cpp#L313) to use loans, meanwhile [Jazzy version](https://github.com/ros2/rmw_fastrtps/blob/jazzy/rmw_fastrtps_cpp/src/publisher.cpp#L307) and [Iron version](https://github.com/ros2/rmw_fastrtps/blob/iron/rmw_fastrtps_cpp/src/publisher.cpp#L307) just the type to be plain. 
Sorry for the inconvenience @Syamkamalnadhc.

**Reply by ghost** [reply]
Hi @JesusPoderoso, thank you for your confirmation.
Could you please provide an information about why humble version has such a limitation but not for Jazzy & Iron?

**Reply by JesusPoderoso** [reply]
Hi @arunjs-dev, that decision was taken by the rmw maintainers, and it is out of the eProsima Fast DDS scope.
